core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1291,11 +1291,11 @@ abstract class RDD[T: ClassTag](
     } else {
       val buf = new ArrayBuffer[T]
       val totalParts = this.partitions.length
-      var partsScanned = 0
+      var partsScanned = 0L
```

Comment:
Fair point, in practice this all but certainly won't happen. Note that this patch was already committed to `master` making this a `Long`. It doesn't hurt and is very very theoretically more correct locally. I suppose I don't think it's worth updating again, but I do not feel strongly about it.